shared/src/libs/types/require-property.ts

```diff
@@ -0,0 +1,4 @@
+type RequireProperty<T extends object, K extends keyof T = keyof T> = T &
+  Pick<T, K>;
```

Comment:
```suggestion
type RequireProperty<T extends object, K extends keyof T = keyof T> = T &
  Required<Pick<T, K>>;
```
otherwise it won't do anything 🤔